pinot-core/src/main/java/org/apache/pinot/core/common/DataFetcher.java

```
@@ -594,4 +594,17 @@ public void close()
       }
     }
   }
+
+  /**
+   * Close the DataFetcher and release all resources (specifically, the ForwardIndexReaderContext off-heap buffers).
+   */
+  public void close() {
+    try {
+      for (ColumnValueReader columnValueReader : _columnValueReaderMap.values()) {
+        columnValueReader.close();
+      }
+    } catch (IOException e) {
+      // do nothing
```

Should we throw the exception out? Swallow exception silently is not good practice unless it is expected

---

Same for other places

---

I think it becomes a backwards incompat change if we rethrow it - changing it to close the context on a jvm that does not support the same cleaner methods would be breaking

Another approach might be to use `CleanerUtil.closeQuietly(..)` within each context's close() impl, and then just use AutoClosable instead of Closable for the ColumnValueReader

---

Using `CleanerUtil.closeQuietly()` might be a good solution. At least we know that is intended

---

Updated to `CleanerUtil.closeQuietly()` and ForwardIndexReaderContext to no longer throw an exception on close as none of the implementations do.